File: Data Structures & Algorithms/eating-bananas/submission-0.js
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles, h) {
        let l = 1, r = Math.max(...piles)
        let result = r
        while(l<=r){
            const k = Math.floor((l+r)/2) 
            let hours = 0
            for(let pile of piles){
                hours+= Math.ceil(pile/k)
            }
            if(hours<=h){
                result = Math.min(result, k)
                r=k-1
            } else{
                l=k+1
            }
        }
        return result
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        result = nums[0]
        for i in range(len(nums)):
            summ=0
            for j in range(i, len(nums)):
                summ+=nums[j]
                result = max(result,summ)
        return result
